src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -0,0 +1,358 @@
+        /// </summary>
+        /// <param name="left">The buffer to compare with <paramref name="right" />.</param>
+        /// <param name="right">The buffer to compare with <paramref name="left" />.</param>
+        /// <returns><see langword="true" /> if the corresponding elements in <paramref name="left" /> and <paramref name="right" /> were equal ignoring case considerations and ASCII. <see langword="false" /> otherwise.</returns>
+        /// <remarks>If both buffers contain equal, but non-ASCII characters, the method returns <see langword="false" />.</remarks>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(left, right);
+
+        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(right, left);
+
+        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(right, left);
+
+        private static bool SequenceEqualIgnoreCase<TLeft, TRight>(ReadOnlySpan<TLeft> left, ReadOnlySpan<TRight> right)
+            where TLeft : unmanaged, INumberBase<TLeft>
+            where TRight : unmanaged, INumberBase<TRight>
+        {
+            Debug.Assert(left.Length == right.Length);
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                uint valueA = uint.CreateTruncating(left[i]);
+                uint valueB = uint.CreateTruncating(right[i]);
+
+                if (!UnicodeUtility.IsAsciiCodePoint(valueA) || !UnicodeUtility.IsAsciiCodePoint(valueB))
+                {
+                    return false;
+                }
+
+                if (valueA == valueB)
+                {
+                    continue; // exact match
+                }
+
+                valueA |= 0x20u;
+                if ((uint)(valueA - 'a') > (uint)('z' - 'a'))
+                {
+                    return false; // not exact match, and first input isn't in [A-Za-z]
+                }
+
+                if (valueA != (valueB | 0x20u))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<ushort> Widen(Vector64<byte> bytes)
+        {
+            if (AdvSimd.IsSupported)
+            {
+                return AdvSimd.ZeroExtendWideningLower(bytes);
+            }
+            else
+            {
+                (Vector64<ushort> lower, Vector64<ushort> upper) = Vector64.Widen(bytes);
```

Comment:
IIRC `Vector64` uses the software fallback on x64.
[crossgen2 with main-branch](https://godbolt.org/z/65bxWG8v9) shows that too.

I don't know if there are plans or WIP to hw-accelerate Vector64 in the .NET 8 timeframe.
Otherwise maybe read the value as long into a Vector128 and do the widening then. Something like
```c#
static Vector128<ushort> LoadAndWiden(ref byte ptr)
{
    if (AdvSimd.IsSupported)
    {
        Vector64<byte> vec64 = Vector64.LoadUnsafe(ref ptr);
        return AdvSimd.ZeroExtendWideningLower(vec64);
    }
    else if (Sse2.IsSupported)
    {
        ulong value = Unsafe.ReadUnaligned<ulong>(ref ptr);
        Vector128<byte> vec = Vector128.CreateScalarUnsafe(value).AsByte();
        return Sse2.UnpackLow(vec, Vector128<byte>.Zero).AsUInt16();
    }
    else
    {
        Vector64<byte> vec64 = Vector64.LoadUnsafe(ref ptr);
        (Vector64<ushort> lower, Vector64<ushort> upper) = Vector64.Widen(vec64);
        return Vector128.Create(lower, upper);
    }
}
```